src/tests/ftest/telemetry/dkey_akey_enum_punch.py

```diff
@@ -0,0 +1,371 @@
+        self.target_per_rank = None
+        self.errors = []
+        self.ioreqs = []
+        self.dkeys_a = []
+        self.dkeys_b = []
+        self.dkey_strs_a = []
+        self.dkey_strs_b = []
+        self.akey_strs = []
+        self.obj_count = 100
+
+    def set_num_targets(self):
+        """Define number of targets from server count and target/server.
+        """
+        server_count = len(self.server_managers[0].hosts)
+        self.target_per_rank = self.params.get(
+            "targets", "/run/server_config/servers/*/")
+        self.total_targets = self.target_per_rank * server_count
+
+    def write_objects_insert_keys(self, container, objtype):
+        """Write objects and insert dkeys and akeys in them.
+
+        Args:
+            container (DaosContainer): Container.
+            objtype (str): Object class.
+        """
+        for i in range(self.obj_count):
+            self.ioreqs.append(IORequest(
+                context=self.context, container=container, obj=None,
+                objtype=objtype))
+
+            # Prepare 2 dkeys and 1 akey in each dkey. Use the same akey for
+            # both dkeys.
+            dkey_str_a = b"Sample dkey A %d" % i
+            dkey_str_b = b"Sample dkey B %d" % i
+            akey_str = b"Sample akey %d" % i
+            data_str = b"Sample data %d" % i
+            self.dkey_strs_a.append(dkey_str_a)
+            self.dkey_strs_b.append(dkey_str_b)
+            self.akey_strs.append(akey_str)
+            data = create_string_buffer(data_str)
+
+            # Pass in length of the key so that it won't have \0 termination.
+            # Not necessary here because we're not interested in the list
+            # output. Just for debugging.
+            self.dkeys_a.append(
+                create_string_buffer(value=dkey_str_a, size=len(dkey_str_a)))
+            self.dkeys_b.append(
+                create_string_buffer(value=dkey_str_b, size=len(dkey_str_b)))
+            akey = create_string_buffer(value=akey_str, size=len(akey_str))
+            c_size = ctypes.c_size_t(ctypes.sizeof(data))
+
+            # Insert the dkeys.
+            self.ioreqs[-1].single_insert(
+                dkey=self.dkeys_a[-1], akey=akey, value=data, size=c_size)
+            self.ioreqs[-1].single_insert(
+                dkey=self.dkeys_b[-1], akey=akey, value=data, size=c_size)
+
+    def punch_all_keys(self):
+        """Punch all dkeys and akeys in the objects.
+        """
+        for i in range(self.obj_count):
+            self.ioreqs[i].obj.punch_akeys(
+                0, dkey=self.dkey_strs_a[i], akeys=[self.akey_strs[i]])
+            self.ioreqs[i].obj.punch_akeys(
+                0, dkey=self.dkey_strs_b[i], akeys=[self.akey_strs[i]])
+            # Punch dkeys one at a time. DAOS-8945
+            self.ioreqs[i].obj.punch_dkeys(0, dkeys=[self.dkey_strs_a[i]])
+            self.ioreqs[i].obj.punch_dkeys(0, dkeys=[self.dkey_strs_b[i]])
+            if i % 10 == 0:
```

Comment:
Is this % 10 just to avoid flooding the logs?

Reply:
Yes. I'm printing it because punch doesn't print anything on job.log.